xdsl/dialects/llvm.py

```diff
@@ -91,6 +92,79 @@ def opaque():
     def typed(type: Attribute):
         return LLVMPointerType([type, NoneAttr()])
 
+    def is_typed(self):
+        return not isinstance(self.type, NoneAttr)
+
+
+@irdl_op_definition
+class GEPOp(Operation):
+    name = "llvm.getelementptr"
+
+    ptr: Annotated[Operand, LLVMPointerType]
+    ssa_indices: Annotated[VarOperand, IntegerType]
+
+    elem_type: OptOpAttr[Attribute]
+
+    rawConstantIndices: OpAttr[DenseArrayBase]
+
+    inbounds: OptOpAttr[UnitAttr]
+
+    result: OpResult
+
+    @staticmethod
+    def get(ptr: SSAValue | Operation,
+            indices: list[int] | None = None,
+            ssa_indices: list[SSAValue | Operation] | None = None,
+            result_type: Attribute | None = None,
+            inbounds: bool = False,
+            ptr_type: Attribute | None = None):
+        # construct default mutable argument here:
+        if indices is None:
+            indices = []
+        # TODO: convert indices to DenseI32ArrayAttr
+
+        i32 = IntegerType(32)
+
+        indices_attr = DenseArrayBase.create_dense_int_or_index(i32, indices)
+
+        # construct default mutable argument here:
+        if ssa_indices is None:
+            ssa_indices = []
+
+        # convert a potential Operation into an SSAValue
+        ptr = SSAValue.get(ptr)
+
+        if ptr_type is None:
+            ptr_type = LLVMPointerType.opaque()
+        else:
+            ptr_type = LLVMPointerType.typed(ptr_type)
+
+        # if no result type was give, infer from pointer type
+        if result_type is None:
+            # ptr is an SSAValue
+            # ptr.typ is the type of the SSAValue (so the LLVMPointerType)
+            assert isinstance(ptr.typ, LLVMPointerType)
+            # ptr.typ.type is the wrapped type of the pointer
+            # use that as the result type
+            result_type = ptr_type
+
+        attrs: dict[str, Attribute] = {
+            'rawConstantIndices': indices_attr,
+            'elem_type': result_type
+        }
+
+        if ptr_type.is_typed():
+            attrs.pop('elem_type')
```

Comment:
I dont like constructing the dict just to remove something right after. There is pribanly python magic that lets you do that in one construction. If not, pleade construct the dict and conditionally add instead of remove.

Reply:
yeah I agree, the dict should be populated gradually

Reply:
Ah yeah, thats more logical